### DISCUSSION TITLE
WARNING - ignoring corrupt image/label: object of type 'int' has no len()

### DISCUSSION THREAD

**Discussion by sangnguyen1006** [ultralytics/ultralytics#7561] [Q&A]
I performed object detection training on my custom dataset using YOLOv8.1.1. After YOLO checked the images and labels in my custom dataset, it gave me a warning just like the screenshot. Please help me explain why it warns me and how to fix it. Thank you very much.

![Untitled](https://github.com/ultralytics/ultralytics/assets/60694260/39016dc7-2d10-45c2-ab05-edff612cc684)

![Untitled1](https://github.com/ultralytics/ultralytics/assets/60694260/10192a2f-3cac-4afc-8a34-f90b7e9ab4aa)

This is one of my labeled files with warnings.

![Untitled3](https://github.com/ultralytics/ultralytics/assets/60694260/37810b34-b6fc-44db-a694-c3f1a3cb5869)

**Comment by pderrenger** [comment]
@sangnguyen1006 the warning you're encountering suggests that there's an issue with one or more of your label files. Specifically, it seems that YOLOv8 is expecting a sequence (like a list or a tuple) for the label content, but it's encountering an integer value instead.

To resolve this issue, please check the following:

1. Ensure that all label files are in the correct format. Labels should be in a text file with one row per object, each containing class x_center y_center width height format, with values normalized between 0 and 1.
2. Verify that there are no empty label files. If an image has no objects, its corresponding label file should be deleted or contain no rows.
3. Check for any label files that may have been saved incorrectly, perhaps with a single integer value or other non-standard content.

You can also use the `--rect` or `--auto-anchor` flags during training to automatically correct some issues with labels. For more detailed guidance on preparing your data and troubleshooting, please refer to our documentation on [Train](https://docs.ultralytics.com/modes/train) mode.

If the problem persists after checking your label files, consider sharing a snippet of the label file content (not as an image) for further assistance. 🛠️
